props/saber_fett263_buttons.h

```diff
@@ -251,10 +265,18 @@ SaberFett263Buttons() : PropBase() {}
 
       // EVENT_PUSH
       if (fabs(mss.x) < 3.0 &&
-          mss.y * mss.y + mss.z * mss.z > 120 &&
+          mss.y * mss.y + mss.z * mss.z > 100 &&
           fusor.swing_speed() < 30 &&
           fabs(fusor.gyro().x) < 10) {
-        Event(BUTTON_NONE, EVENT_PUSH);
+          if (!push_begin_) {
```

Comment:
I don't think this works that well.
Basically, when pushing is detected, it starts a timer, and when that timer expires, the next time pushing is detected, a push event is generated.

I think what you want is something like this:

    if ( DETECT PUSH ) {
       if (millis() - push_begin_millis_ > FETT263_PUSH_LENGTH) {
         Event(BUTTON_NONE, EVENT_PUSH);
         push_begin_millis_ = millis();
       }
     } else {
        push_begin_millis_ = millis();
     }

Basically, push_begin_millis_ is set to millis every time a push is NOT detected. That way, millis() - push_begin_millis_ will only become 25 if push was detected every time we get to this code for 25 milliseconds.

Same thing goes for thrust below.

Reply:
thanks, I was thinking the push_begin_millis_ would be continuously overwritten if I didn't include a bool, I've updated both the same, it's working for me but the target for this is the users with the speaker vibrations so will need their testing to confirm